src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -490,21 +491,16 @@ private void initialize() {
         }
     }
 
-    private Optional<BitbucketCommit> fetchBitbucketCommit(BitbucketBranchSCMHead head) {
+    private Optional<BitbucketCommit> fetchBitbucketCommit(BitbucketSCMHead head) {
         return getScmHelper().map(scmHelper -> scmHelper.getCommitClient(getProjectKey(), getRepositorySlug())
-                .getCommit(head.getName()));
+                .getCommit(head.getFullRef()));
```

Comment:
This changes how both branch and tags work- have we tested this in both cases?

Reply:
both have been manually tested, and the automatic testing has been updated to work with this method.